block.go

```diff
@@ -649,9 +649,11 @@ func isFenceLine(data []byte, info *string, oldmarker string, newlineOptional bo
 	}
 
 	i = skipChar(data, i, ' ')
-	if i >= len(data) || data[i] != '\n' {
+	if l := len(data); i >= l || data[i] != '\n' {
```

Comment:
Somewhat minor nit: could you please change the name of `l` to something else? It's too similar to `i` and `1` and is difficult to read.

Also, this entire chain of conditionals is becoming very convoluted. A pattern like

```go
if cond ... {
  return X
} else if !cond ... {
  also return X
}
```

is a sure sign this should be simplifiable. I'll try to get back with a more concrete suggestion, but just leaving this note here for now.